tls/options.go

```diff
@@ -107,23 +110,29 @@ func startCAReloader(ctx context.Context, g run.Group, upstreamCAFile string, in
 	return caReloader, nil
 }
 
+// hasCA determine if the CA was specified.
 func (uo *UpstreamOptions) hasCA() bool {
 	return len(uo.ca) != 0 || uo.caReloader != nil
 }
 
+// hasCA determine if the hasUpstreamCerts were specified.
 func (uo *UpstreamOptions) hasUpstreamCerts() bool {
 	return uo.cert != nil || uo.certReloader != nil
 }
 
+// hasCA determine if the CA watcher is enabled.
 func (uo *UpstreamOptions) isCAReloadEnabled() bool {
 	return uo.caReloader != nil
 }
 
+// hasCA determine if the certificate watcher is enabled.
```

Comment:
Looks like something went wrong with the doc of these methods here.

Reply:
Fixed! Thanks!